indigo/bulk_creator.py

```diff
@@ -382,19 +381,26 @@ def link_publication_document(self, work, info):
         pub_doc.save()
 
     def link_commencement(self, work, info):
-        # if the work is `commenced_by` something, try linking it
-        # make a task if this fails
-        title = info['commenced_by']
-        work = info['work']
-        commencing_work = self.find_work_by_title(title)
-        if not commencing_work:
-            return self.create_task(work, info, task_type='link-commencement')
-
-        work.commencing_work = commencing_work
-        try:
-            work.save_with_revision(self.user)
-        except ValidationError:
-            self.create_task(work, info, task_type='link-commencement')
+        # if the work has commencement details, try linking it
+        # make a task if a `commenced_by` title is given but not found
+        date = info.get('commencement_date')
+
+        commencing_work = None
+        title = info.get('commenced_by')
+        if title:
+            commencing_work = self.find_work_by_title(title)
+            if not commencing_work:
+                self.create_task(work, info, task_type='link-commencement')
+
+        commencement = Commencement(
+            commenced_work=work,
+            commencing_work=commencing_work,
+            date=date,
```

Comment:
Is it okay if `date` is None?

Reply:
yep, the date is allowed to be None: 
`date = models.DateField(null=True, blank=True, help_text="Date of the commencement, or null if listed provisions are uncommenced")`

Reply:
we're obviously not _expecting_ a work to have a commencing work but no date on a bulk import, but if it does happen the work will be flagged as 'commencement date unknown' and we take it from there